iceoryx_binding_c/source/c_subscriber.cpp

```diff
@@ -102,18 +106,32 @@ iox_sub_t iox_sub_init(iox_sub_storage_t* self,
         subscriberOptions.requiresPublisherHistorySupport = options->requirePublisherHistorySupport;
     }
 
+    // this is required for CycloneDDS to limit the fallout of our change to use the heap for storage
+    // it should be removed with #1221
+    auto meWithStoragePointer = new SubscriberWithStoragePointer();
+    meWithStoragePointer->subscriberStorage = self;
+    auto me = &meWithStoragePointer->subscriber;
+    assert(reinterpret_cast<uint64_t>(me) - reinterpret_cast<uint64_t>(meWithStoragePointer) == sizeof(void*)
```

Comment:
OK, we need this to be sure our left shift by 8 bytes to get the pointer will work. Not sure if we can conclude that if it is true in debug mode, it will also hold in release mode. We could use an `Ensures` to be safer but maybe we want to take the risk.

Reply:
This is just my inner chicken. The C standard actually requires this to be true for a struct with this two members.